NDTensors/src/abstractarray/similar.jl

```diff
@@ -56,6 +56,9 @@ function similar(arraytype::Type{<:AbstractArray}, dims::Tuple)
   return similartype(arraytype, shape)(undef, NDTensors.to_shape(arraytype, shape))
 end
 
+# For when there are CUArray specific issues inline
+iscu(A::AbstractArray) = iscu(leaf_parenttype(A))
+iscu(::Type{<:AbstractArray}) = false
```

Comment:
An issue I see here is that if you pass a type with a nested structure, say `iscu(Adjoint{Float64,CuMatrix{Float64}})`, it looks like this would return `false`.

We could do:
```julia
iscu(A::AbstractArray) = iscu(typeof(A))
iscu(arraytype::Type{<:AbstractArray}) = iscuarray(leaf_parenttype(arraytype))
iscuarray(::Type{<:AbstractArray}) = false
```
and then:
```julia
iscuarray(::Type{<:CuArray}) = true
```